rfcs/InputUnion.md

```diff
@@ -149,6 +149,13 @@ There have been a variety of use cases described by users asking for an abstract
 * [Observability Cloud Integrations](https://gist.github.com/binaryseed/f2dd63d1a1406124be70c17e2e796891#cloud-integrations)
 * [Observability Dashboards](https://gist.github.com/binaryseed/f2dd63d1a1406124be70c17e2e796891#dashboards)
 
+## Prior art
+
+Other technologies/languages also faced the same problems and there things we can learn from them:
+
+* [Wiki article on Tagged_union](https://en.wikipedia.org/wiki/Tagged_union)
```

Comment:
[TypeScript Discriminated Unions](http://www.typescriptlang.org/docs/handbook/advanced-types.html#discriminated-unions) are an interesting example of the tagged union pattern. They're not fundamentally different than the other examples discussed in the Tagged Union Wikipedia article but it's interesting to see how they use string literal types to avoid ambiguity in their structural type system (and it's a language that large parts of the GraphQL community are familiar with).